lib/spack/spack/report.py

```diff
@@ -247,30 +260,196 @@ class collect_info(object):
     Raises:
         ValueError: when ``format_name`` is not in ``valid_formats``
     """
-    def __init__(self, specs, format_name, filename):
-        self.specs = specs
+    def __init__(self, format_name, install_command):
         self.format_name = format_name
+        # Consider setting these properties in a more CDash specific place.
+        self.install_command = install_command
+        self.hostname = socket.gethostname()
+        self.osname = platform.system()
+        self.starttime = int(time.time())
+        self.buildstamp = time.strftime("%Y%m%d-%H%M-Experimental",
+                                        time.localtime(self.starttime))
 
         # Check that the format is valid
         if format_name not in itertools.chain(valid_formats, [None]):
             raise ValueError('invalid report type: {0}'.format(format_name))
 
-        self.filename = filename
-        self.collector = InfoCollector(specs) if self.format_name else None
-
     def __enter__(self):
         if self.format_name:
             # Start the collector and patch PackageBase.do_install
+            self.collector = InfoCollector(self.specs)
             self.collector.__enter__()
 
+    def cdash_initialize_report(self, report_data):
+        if not os.path.exists(self.filename):
+            os.mkdir(self.filename)
+        report_data['install_command'] = self.install_command
+        report_data['buildstamp'] = self.buildstamp
+        report_data['hostname'] = self.hostname
+        report_data['osname'] = self.osname
+
+    def cdash_build_report(self, report_data):
```

Comment:
I believe this code belongs to an external class. The rationale is that before the PR `collect_info` was collecting generic information during the build and passing a dictionary to a template engine. It's fine if the additional information the PR brings in is only used by CDash, but now it seems to me there's a lot of code that is CDash specific in the collector itself.